### ISSUE TITLE
false positive Incompatible types in assignment for overloaded function

### ISSUE THREAD

**Issue by sam-s** [python/mypy#21784]
I declare a variable type and get **assignment** error for [`gzip.open`](https://docs.python.org/3/library/gzip.html#gzip.open) but not for [`open`](https://docs.python.org/3/library/functions.html#open),

**To Reproduce**

```python
from collections.abc import Callable
from typing import IO
import gzip

foo : Callable[[str, str], IO] = gzip.open
```

**Expected Behavior**

no diagnostics

**Actual Behavior**

> error: Incompatible types in assignment (expression has type overloaded function, variable has type "Callable[[str, str], IO[Any]]")  [assignment]

but no diagnostics for
```python
from collections.abc import Callable
from typing import IO

foo : Callable[[str, str], IO] = open
```

**Your Environment**

- Mypy version used: `mypy 2.3.0`
- Mypy command-line flags: `--check-untyped-defs`
- Python version used: `3.14.6`

**Comment by sam-s** [comment]
Alternatively:
```python
import gzip

def open_ext(compress:bool):
    if compress:
        o = gzip.open
        e = ".gz"
    else:
        o = open
        e = ""
    return (o,e)
```
results in
> error: Incompatible types in assignment (expression has type overloaded function, variable has type overloaded function)  [assignment]

**Comment by ilevkivskyi** [comment]
If you think there is a problem with builtin stubs you should report this to typeshed instead.
